src/error.rs

```diff
@@ -159,19 +163,8 @@ impl Error {
     }
 
     /// Returns the labels for this error.
-    pub fn labels(&self) -> &[String] {
-        match self.kind.as_ref() {
-            ErrorKind::Command(ref err) => &err.labels,
-            ErrorKind::Write(ref err) => match err {
-                WriteFailure::WriteError(_) => &self.labels,
-                WriteFailure::WriteConcernError(ref err) => &err.labels,
-            },
-            ErrorKind::BulkWrite(ref err) => match err.write_concern_error {
-                Some(ref err) => &err.labels,
-                None => &self.labels,
-            },
-            _ => &self.labels,
-        }
+    pub fn labels(&self) -> &HashSet<String> {
```

Comment:
I changed this to a `HashSet` since ordering doesn't matter, we don't want duplicates, and generally only care about whether a label is present or not. Arguably we could just not expose this at all given that we have `contains_label` already but I figured we already had it so might as well keep it.